### PR TITLE
Feature :: Whitebit new public API

### PR DESCRIPTION
**Pull request by carlosmiei**
The goal of the PR is to migrate the current public part of Whitebit to the new v4 version, however, since this version does not have feature parity yet I've decided to migrate only what is already available in the latest version to avoid losing too much information.

**Newly added:**

- fetchTime
- fetchFundingFees

**Migrated:**

- FetchCurrencies
- FetchOrderBook
- FetchTrades (removed fetchTradesV1 and fetchTradesV2)
- FetchStatus
- FetchTickers

**Stayed as-is (lacking in v4):**

- FetchMarket
- FetchTicker
- FetchTradingFees
- fetchOHLCV

### CODE REVIEW

**Comment by kroitor** [comment]
Please, check this test error: https://app.travis-ci.com/github/ccxt/ccxt/builds/243965824#L7490

File: js/whitebit.js [hunk 1]
```diff
@@ -361,103 +401,68 @@ module.exports = class whitebit extends Exchange {
...
-        const last = this.safeNumber (ticker, 'last');
-        const percentage = this.safeNumber (ticker, 'change');
-        let change = undefined;
-        if (percentage !== undefined) {
-            change = this.numberToString (percentage * 0.01);
-        }
+        const last = this.safeNumber (ticker, 'last_price');
+        const percentage = this.safeNumber (ticker, 'change') * 0.01;
+        const change = last * percentage;
+        const initial = last - change;
```
Comment (kroitor) [inline js/whitebit.js:421]: Let's not calculate lines 418-421, they must be done in `safeTicker`.
We should `return this.safeTicker ({ ... })` – see how fetchTicker / parseTicker is implemented across the other files.

File: js/whitebit.js [hunk 2]
```diff
@@ -267,36 +267,31 @@ module.exports = class whitebit extends Exchange {
...
+        //     "can_withdraw": true,
+        //     "can_deposit": true,
+        //     "min_withdraw": "0.001",
+        //     "max_withdraw": "2",
+        //     "maker_fee": "0.1",
+        //     "taker_fee": "0.1",
+        //     "min_deposit": "0.0001",
+        //     "max_deposit": "0",
+        //  },
+        // }
```
Comment (kroitor) [inline js/whitebit.js:284]: I'd really appreciate, if you could follow the comment formatting precisely, compare formatting, indentation, whitespaces on the left to formatting on the right:

Screenshot 2021-12-24 at 13 19 12

Comment (carlosmiei) [reply js/whitebit.js:284]: @kroitor  sorry, I'll fix it

File: js/whitebit.js (hunk 1 above)
Comment (carlosmiei) [reply js/whitebit.js:421]: @kroitor lines 418 (last) and 419 (percentage), I'm not calculating them only parsing from the raw ticker, those I can keep right?

Comment (kroitor) [reply js/whitebit.js:421]: Yep.

File: js/whitebit.js [hunk 3]
```diff
@@ -598,22 +563,13 @@ module.exports = class whitebit extends Exchange {
         //         "deal": "0.00419198" // amount in money
         //     }
         //
-        let timestamp = this.safeTimestamp (trade, 'time');
-        if (timestamp === undefined) {
-            timestamp = this.parse8601 (this.safeString (trade, 'time'));
-        }
+        const timestamp = this.safeTimestamp2 (trade, 'time', 'trade_timestamp');
         const orderId = this.safeString (trade, 'dealOrderId');
         const cost = this.safeString (trade, 'deal');
```
Comment (kroitor) [inline js/whitebit.js:568]: I think cost is also the quote_volume here? Should we use safeString2 ? Does the math hold?

**Pull request merged**